Don't drop leading zeroes when performing generic ECDSA signing

hslatman opened the pull request:
call ret.FillBytes() instead of ret.Bytes() to preserve leading zeroes that may have been dropped when converting the digest to an integer

Cherry-picked from https://github.com/google/go-attestation/pull/357

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/smallstep/go-attestation?pullRequest=6) <br/>All committers have signed the CLA.

hslatman committed:
Don't drop leading zeroes when performing generic ECDSA signing

call ret.FillBytes() instead of ret.Bytes() to preserve leading zeroes that may have been dropped when converting the digest to an integer
